### ISSUE TITLE
Page creation and file attachment does not handle deep paths (length >260)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [fracte/keepnote#162]
```
What steps will reproduce the problem?
1. Create a hierarchy of notes such that their ultimate path is close to
260 characters long.
2. Try and attach a file to a note at the bottom of the chain with a name
which will make the path longer then 260 characters.
3. Attachment will fail since the set_payload function uses full-file paths
and shutil can't see the destination (or source, or both depending on other
path lengths).

What is the expected output? What do you see instead?
The expected behavior would be not to see this issue at all since unicode
filenames can be up to 32,000 characters long. The set_payload function
(and I suspect others) should intelligently use relative paths so as not to
exceed 260 characters (for Windows - there is the \\?\ notation which
allows windows functions to succeed but may not be Linux compatible).

What version of the product are you using? On what operating system?
Windows 7.

Please provide any additional information below.
I discovered this while writing an import plugin to pull in folder trees to
Keepnote in order to store a large complicated chain of data. It does not
appear fixable on just the extension side.
```

Original issue reported on code.google.com by `W.Roues...@gmail.com` on 1 Dec 2009 at 1:46

**Comment by GoogleCodeExporter** [comment]
```
Can you send the exact error message.  I would like to which function is giving 
the
error.  The error should be in the error log "Help > Show Error Log".
Thanks,
Matt
```

Original comment by `matt.rasmus@gmail.com` on 2 Jan 2010 at 8:38

**Comment by GoogleCodeExporter** [comment]
```
The exact error message for a typical case is as follows. Note that the file 
most
definitely exists when I did this:

KeepNote 0.6: Tue Dec 01 18:42:32 2009
Traceback (most recent call last):
  File "C:\Users\Will
Rouesnel\Documents\Development\keepnote-0.6-will\src\keepnote\extensions\import_
folders\__init__.py",
line 99, in <lambda>
    window.get_notebook())),
  File "C:\Users\Will
Rouesnel\Documents\Development\keepnote-0.6-will\src\keepnote\extensions\import_
folders\__init__.py",
line 134, in on_import_folder_tree
    self.import_folder_tree(notebook, filename, window=window, widget=widget)
  File "C:\Users\Will
Rouesnel\Documents\Development\keepnote-0.6-will\src\keepnote\extensions\import_
folders\__init__.py",
line 138, in import_folder_tree
    import_folder(notebook, filename, window=window, task=None, widget=widget)
  File "C:\Users\Will
Rouesnel\Documents\Development\keepnote-0.6-will\src\keepnote\extensions\import_
folders\__init__.py",
line 273, in import_folder
    child.set_payload(os.path.join(root, file))
  File "C:\Users\Will
Rouesnel\Documents\Development\keepnote-0.6-will\src\keepnote\notebook\__init__.
py",
line 1086, in set_payload
    raise NoteBookError(_("Cannot copy file '%s'" % filename), e)
keepnote.notebook.NoteBookError: IOError(2, 'No such file or directory')
Cannot copy file 'C:\Users\Will Rouesnel\Documents\University\PhD
2008\Experiments\.backup\0.Gold nanocube electrochemical synthesis\TEM
Images\Experiment 1 Repeat 2\Experiment 1 Rpt 2 - Au, CTAB+TTAB, 1.6 mA - 2.dm3'
```

Original comment by `W.Roues...@gmail.com` on 14 Jan 2010 at 3:27

**Comment by GoogleCodeExporter** [comment]
```
One test you could do for me to see if the \\?\ notation works is to get 
keepnote to
open your notebook with \\?\ notation.  Then all paths afterwards will have 
\\?\ as
its prefix.  One way to do that is to edit the config file to change the file 
path of
your notebook in the recently open notebooks section.  Using a text editor (like
wordpad) open C:\\Documents and Settings\USERNAME\Application 
Data\takenote\takenote.xml

Look for the file name of your notebook within the xml and add the \\?\ 
notation to
the beginning.  Then run keepnote and use the recently opened notebooks menu to 
open
your notebook.  See if file operations work better that way.  If this works, I 
might
add a work around to keepnote to prefix this notation all the time when running 
on
windows.
```

Original comment by `matt.rasmus@gmail.com` on 18 Jan 2010 at 4:35

**Comment by GoogleCodeExporter** [comment]
```
I get an error too when I try to attach a file which has a relatively long name
including non-English characters. I've attached the error log.

```

Original comment by `ertugrul...@gmail.com` on 10 Mar 2010 at 10:35

Attachments:
- [error-log.txt.bak.rtf](https://storage.googleapis.com/google-code-attachments/keepnote/issue-162/comment-4/error-log.txt.bak.rtf)

**Comment by GoogleCodeExporter** [comment]
```
Ok I've tried changing the path to \\?\ - this does lead to the obvious issue 
that
this notation will not work outside of Windows.

But - it works. All the paths have \\?\ appended to them and with the 
appropriate
change added to my "Folder Tree" import extension I can pull in very deep paths 
and
manipulate them just fine, which is great.
```

Original comment by `W.Roues...@gmail.com` on 8 Apr 2010 at 2:47

**Comment by GoogleCodeExporter** [comment]
Original comment by `matt.rasmus@gmail.com` on 15 Jun 2010 at 5:58
- Added labels: **win**

**Comment by GoogleCodeExporter** [comment]
```
Issue 332 has been merged into this issue.
```

Original comment by `matt.rasmus@gmail.com` on 7 Nov 2010 at 4:03

**Comment by GoogleCodeExporter** [comment]
```
Issue 452 has been merged into this issue.
```

Original comment by `matt.rasmus@gmail.com` on 25 Apr 2011 at 3:28

**Comment by GoogleCodeExporter** [comment]
```
This is now fixed in testing version keepnote 0.7.2
http://keepnote.org/keepnote/download-test/

```

Original comment by `matt.rasmus@gmail.com` on 25 Apr 2011 at 8:09
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
This is now fixed in 0.7.2.
```

Original comment by `matt.rasmus@gmail.com` on 14 May 2011 at 7:05
- Changed state: **Done**
